packages/vite/src/utils/generator-utils.ts

```diff
@@ -497,3 +566,106 @@ export function getViteConfigPathForProject(
 
   return normalizeViteConfigFilePathWithTree(tree, root, viteConfigPath);
 }
+
+export async function handleUnsupportedUserProvidedTargets(
+  unsupportedUserProvidedTarget?: {
+    build?: boolean;
+    serve?: boolean;
+    test?: boolean;
+  },
+  userProvidedTargets?: {
+    build: string;
+    serve: string;
+    test: string;
+  },
+  targets?: {
+    build: string;
+    serve: string;
+    test: string;
+  }
+) {
+  if (unsupportedUserProvidedTarget.build && targets.build) {
```

Comment:
Should this be `unsupportedUserProvidedTarget?.target` and `targets?.build`? Since they are declared as optional args.